src/main/kotlin/com/ampnet/reportservice/service/impl/TemplateDataServiceImpl.kt

```diff
@@ -189,4 +191,9 @@ class TemplateDataServiceImpl(
         userUuid: String
     ): List<Transaction> =
         userTransactions[userUuid]?.mapNotNull { TransactionFactory.createTransaction(it) }.orEmpty()
+
+    private fun throwExceptionIfNoUserWithInfo(users: UsersExtendedResponse) {
+        if (users.usersList.isEmpty())
+            throw ResourceNotFoundException(ErrorCode.USER_MISSING_INFO, "No user has went through kyc on the platform")
```

Comment:
This exception will return BadRequest - 400, but we want Not Found - 404.
Message: "None of the users went through KYC and completed at least one transaction."